go-controller/pkg/cni/helper_linux.go

```diff
@@ -252,6 +253,19 @@ func setupInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInter
 			ifnameSuffix = fmt.Sprintf("_%d", containerVeth.Index)
 		}
 
+		if ifName == "ovn-udn1" {
```

Comment:
uf I don't like how this is done ... let's see if we can come up with a better way. 

For now, invest in the unit tests instead. Let's just keep thinking.

Reply:
note that if we use bridge, we can create the table if it doesnt exists already, and then it doesnt need the ifname 

(we can even create bridge table on the OVN side where LLC exists if desired, in case it simplify things, if we will have the right network NS there)

Reply:
> note that if we use bridge, we can create the table if it doesnt exists already, and then it doesnt need the ifname

That's a good argument for using bridge instead ...

> we can even create bridge table on the OVN side where LLC exists if desired, in case it simplify things, if we will have the right network NS there

Tapping into net namespaces is very expensive IIRC. We should try to do everything that requires accessing the net ns from a single place. 

I would prefer if you pushed the link local address to the CNI layer via network selection elements CNI args for IPv6 networks.

Reply:
> ah oops, its not correct, we still need to create it only if we have some indication that we use primary UDN
which is what this condition also doing (please correct me if i am wrong)

That's what I really dislike about this comparison. You use the name of the interface to infer if primary UDN is used. 

Try to use something like https://github.com/ovn-kubernetes/ovn-kubernetes/blob/30585c17728872c1a1a6d199c3130998ac0f269c/go-controller/pkg/cni/cni.go#L174 (or pass along that value)

Reply:
@maiqueb @qinqon 
please let me know if you want CNI args instead so i can focus on one of the directions

Reply:
I think if you use CNI args you'll do less in the CNI sandbox, which is better. You'll just consume the value previously made available by the ovnkube controller.

But how will you then know if you're in primary UDN ? Would it be the presence of the LRP LLA in the CNI args ?

Reply:
@maiqueb so the idea is to not spread code since this is going to be deleted after transit routers from OVN, so I think is better if we keep everything under CNI, so we don't have to change expectations from CNI args or change other packages.

Reply:
@qinqon I understand the motive, but I don't buy the argument. 

Wouldn't we just revert the commit / whole PR once that lands ? (really sorry Or ...)

Reply:
Since we are going to revert it anyhow once the real fix is out, and since we also had an agreement about the current design, please consider to just keep going with the current one, time is also precious.
Moreover I also think that changing CNI args is more intrusive a little fwiw

Reply:
isn't it easier to unit test the solution with CNI_ARGS ?

Reply:
possible it is, but not sure by heart yet
generally speaking, unit tests will be enough ? (i am against over testing personally)

Reply:
as concluded the annotation on the pod was added on the pending PR, and there will be a bool to add it only if primary exists